lib/improver/nbhood/recursive_filter.py

```diff
@@ -92,23 +92,20 @@ def __init__(self, alpha_x=None, alpha_y=None, iterations=None,
         """
         alpha_error = ("alpha must be less than 0.5. A large alpha value"
                        "leads to poor conservation of probabilities: ")
-        if alpha_x is not None and not 0 < alpha_x <= 0.5:
-            message = alpha_error if alpha_x > 0.5 else ''
-            message += "Invalid alpha_x: must be > 0 and <= 0.5: {}"
-            raise ValueError(message.format(alpha_x))
-        if alpha_y is not None and not 0 < alpha_y <= 0.5:
-            message = alpha_error if alpha_y > 0.5 else ''
-            message += "Invalid alpha_y: must be > 0 and <= 0.5: {}"
-            raise ValueError(message.format(alpha_y))
+        for k, v in {'x': alpha_x, 'y': alpha_y}.items():
```

Comment:
Sorry, one last nitpick - can we please use better variable names than `k`, and `v` please?
e.g: `direction, alpha` or just anything slightly more specific.

Reply:
`i, j`?

just kidding.
Fixed.

Reply:
Thanks